rpc/wrtc_client.go

```diff
@@ -152,6 +155,15 @@ func dialWebRTC(
 		}
 	}()
 
+	var callUpdates int
+	var maxCallUpdateDuration, totalCallUpdateDuration time.Duration
+	onICEConnected := func() {
+		averageCallUpdateDuration := totalCallUpdateDuration / time.Duration(callUpdates)
+		// TODO: Potentially report these stats to sentry/some central location at some point.
+		logger.Debugw("ICE connected", "time_since_dial_start", time.Since(dialStart), "num_call_updates",
+			callUpdates, "average_duration", averageCallUpdateDuration, "max_call_update_duration", maxCallUpdateDuration)
```

Comment:
Our wrtc wrappers + Pion never seem to reach a state of ICE ["completed"](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/iceConnectionState#completed), only "connected". In the flutter SDK, we only report caller updates after the ICE process has been completed, which lets us gather stats on something like 10 caller updates. Here in go, we usually only see 2 or 3 until reaching connected. Still probably worth logging, though.

Reply:
do we never reach "connected" because we use trickle ice?

Reply:
Discussed a bit offline; it's probably not because of trickle ICE. The flutter SDK/`flutter-webrtc` is able to declare a state of ICE completed when trickle ICE is being used. It's probably because our golang WebRTC clients never declare the ICE exchange as "complete" (not sure what this looks like in code). Per @dgottlieb's suggestion, I'm going to add a delay of 5s after we reach ICE connected before reporting these stats in the hope that the stats will include information from more caller updates.

Reply:
I can't just `time.Sleep(5 * time.Second)`, as I think tests would try to log after completion if I did. `select`ed on a 5s timer and the `dialWebRTC` context instead; might be the best we can do unless others have ideas.